### ISSUE TITLE
[Question] Can I modify the license to use AGPL as the future license instead of MIT/Apache 2.0?

### ISSUE THREAD

**Issue by Orenoid** [getsentry/fsl.software#53]
I'm new to software licensing, sorry if I ask any stupid questions.  

I'm currently developing a product and would like to apply the Fair Source License.   
However, I'm reluctant to switch to MIT or Apache 2.0 after the initial period, so I modified the license terms to use AGPL as the future license in my [LICENSE](https://github.com/Orenoid/BabelDuck?tab=License-1-ov-file) file.  

Is this modification legally valid and effective?

**Comment by voxpelli** [comment]
I think https://github.com/getsentry/fsl.software/issues/49#issuecomment-2343706359 and #46 can provide some insight

**Comment by ljharb** [comment]
tldr, no, the point of FSL is that it ends up *free* after the time period. If you want AGPL, you should start with AGPL.

**Comment by Orenoid** [comment]
> I think [#49 (comment)](https://github.com/getsentry/fsl.software/issues/49#issuecomment-2343706359) and [#46](https://github.com/getsentry/fsl.software/issues/46) can provide some insight

Thanks, I should've checked them, sorry about that. I guess the answer is no then, since copyleft licenses are not acceptable.  
If I still would like to apply AGPL after 2 years, given that it doesn't meet the requirements of Fair Source License any more, should I rename the license to something else? Or even can't I do the renaming?

**Comment by Orenoid** [comment]
> tldr, no, the point of FSL is that it ends up _free_ after the time period. If you want AGPL, you should start with AGPL.

Sorry, I'm a bit confused. As @mon-jai had mentioned, AGPL is also free, except it's copyleft.

**Comment by mon-jai** [comment]
> except it's copyleft

Copyleft licenses are considered free by both FSF and OSI.

**Comment by gavin-zee** [comment]
FSL was drafted with the specific intent of converting into permissive open source licenses (not copyleft), because permissive licenses aligned most closely with how we ideally wanted to license our own products when converting. We also wanted to avoid license proliferation, which in our view was one of the biggest downsides of the BUSL. Because any change license could be designated under the BUSL, it made it very difficult from a compliance perspective, as each version was different.

TLDR, FSL is narrow in scope for a reason.

You're free to create your own license based on the FSL with AGPL as the change license if you rename it so that it isn't confusingly similar.

**Comment by voxpelli** [comment]
> You're free to create your own license based on the FSL with AGPL as the change license if you rename it so that it isn't confusingly similar.

@gavin-zee What's the license of the FSL license?

Eg. BSL says `In consideration of the right to use this License’s text [...] Not to modify this License in any other way.` in the `Covenants of Licensor` at https://mariadb.com/bsl11/

Also the FSF has some wordings in regards to modifying the GPL licenses: https://www.gnu.org/licenses/gpl-faq.html#ModifyGPL

Maybe the FSL should also have some wording? And since the FSL is narrow in scope for a reason, it would probably be better for most who feel its too narrow to use the BSL rather than fork the FSL license? Should probably be added to the FAQ as well?

**Comment by gavin-zee** [comment]
@voxpelli we wanted to keep the FSL simple so we avoided the license the license language. That said, we can consider addressing in the FAQs since this question of changes to the license has come up a few times now. (cc @chadwhitacre)

**Comment by voxpelli** [comment]
At a minimum I think it should say that when using the FSL name no modifications are allowed

**Comment by Orenoid** [comment]
Based on the discussion above, I assume I should rename the license to differentiate it from FSL. 
Thanks for all the clarification and help. Feel free to close the issue.

**Comment by chadwhitacre** [comment]
> consider addressing in the FAQs
> At a minimum I think it should say that when using the FSL name no modifications are allowed
> 

Is https://github.com/getsentry/fsl.software/pull/54 good enough?

**Comment by voxpelli** [comment]
With the addition from #54 I think this issue is answered and can be closed

**Comment by chadwhitacre** [comment]
Thanks all. :)
